src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts

```diff
@@ -1327,21 +1333,26 @@ export class TerminalTaskSystem extends Disposable implements ITaskSystem {
 
 	private _reconnectToTerminals(): void {
 		if (this._hasReconnected) {
+			this._logService.trace(`Already reconnected, to ${this._reconnectedTerminals?.length} terminals so returning`);
 			return;
 		}
 		this._reconnectedTerminals = this._terminalService.getReconnectedTerminals(ReconnectionType)?.filter(t => !t.isDisposed);
+		this._logService.trace(`Attempting reconnection of ${this._reconnectedTerminals?.length} terminals`);
 		if (!this._reconnectedTerminals?.length) {
+			this._logService.trace(`No terminals to reconnect to so returning`);
 			return;
 		}
 		for (const terminal of this._reconnectedTerminals) {
 			const task = terminal.shellLaunchConfig.attachPersistentProcess?.reconnectionProperties?.data as IReconnectionTaskData;
+			this._logService.trace(`Reconnecting to task: ${JSON.stringify(task)}`);
```

Comment:
It's best not to perform work like stringifying here when logging as it will happen even if it's not logged. This is especially important on start up.

```suggestion
			if (getLevel() <= LogLevel.Trace) {
				this._logService.trace(`Reconnecting to task: ${JSON.stringify(task)}`);
			}
```

`task` is also a class so it will stringify a lot, you might want to split that out to only log the parts you care about?